tests/register/register.go

```diff
@@ -31,6 +38,68 @@ func init() {
 	Tests = make(map[TestType][]types.Test)
 }
 
+var count = 0
+
 func Register(tType TestType, t types.Test) {
 	Tests[tType] = append(Tests[tType], t)
+	// count += howmany(t.Config)
+	count++
+	fmt.Printf("count now at: %d\n", count)
+}
+
+// Finds version, creates Test structs for all compatible versions and registers them
+func RegisterTest(tType TestType, t types.Test) {
+	versions := [][]string{
+		{"0.0.0"},
+		{"1.0.0"},
+		{"2.0.0", "2.1.0", "2.2.0", "2.3.0-experimental"},
+	}
+
+	pattern := regexp.MustCompile("[0-9]{1}.[0-9]{1}.[0-9]{1}")
+	versionArr := pattern.FindAllStringSubmatch(t.Config, 1)
+	if len(versionArr) == 0 { // handles negative/general/invalid_version.go
+		Register(tType, t)
+		return
+	}
+	version := versionArr[0][0]
+
+	major_version, _ := strconv.Atoi(version[0:1])
+	minor_version, _ := strconv.Atoi(version[2:3])
+	if major_version == 4 { // handles negative/general/config.go
+		Register(tType, t)
+		return
+	}
+
+	Register(tType, t) // register original test
+
+	fmt.Printf("Config version: %s; %d total test; ", version, len(versions[major_version])-minor_version)
+	for i := minor_version; i < len(versions[major_version])-1; i++ {
+		test_tmp := t
+
+		test_tmp.Config = strings.Replace(test_tmp.Config, versions[major_version][i], versions[major_version][i+1], 1)
+
+		// for i := 0; i < len(t.In); i++ {
+		// 	disk := t.In[i]
+		// 	for j := 0; j < len(disk.Partitions); j++ {
+		// 		partition := disk.Partitions[j]
+		// 		tmp_partition := new(types.Partition)
+		// 		*tmp_partition = *partition
+
+		// 		test_tmp.In[i].Partitions[j] = tmp_partition
+		// 	}
+		// }
+
+		// for i := 0; i < len(t.Out); i++ {
+		// 	disk := t.In[i]
+		// 	for j := 0; j < len(disk.Partitions); j++ {
+		// 		partition := disk.Partitions[j]
+		// 		tmp_partition := new(types.Partition)
+		// 		*tmp_partition = *partition
+
+		// 		test_tmp.Out[i].Partitions[j] = tmp_partition
+		// 	}
+		// }
+
+		Register(tType, test_tmp)
```

Comment:
The version number being used should be appended to the tests. Otherwise we have no way of easily deciphering at a glance what the version is (the tests show up as `ExampleTest#01, ExampleTest#02, ...`)

Reply:
Sounds good. Will update